### PR TITLE
2863 implement new how do i earn page

### PR DESCRIPTION
**Pull request by msquance-stem**
## Status

* Current Status: Ready for front-end / tech review
* Review App link(s): https://teachcomputing-pr-2231.herokuapp.com/how-do-i-earn
* Closes: https://github.com/NCCE/teachcomputing.org-issues/issues/2863

## Review progress:

- [ ] Browser tested
- [ ] Front-end review completed
- [ ] Tech review completed

## What's changed?

*Description of what's been done - bullets are usually best, but you do you.*

## Steps to perform after deploying to production

*If the production environment requires any extra work after this PR has been deployed detail it here. This could be running a Rake task, migrating a DB table, or upgrading a Gem. That kind of thing.*

### CODE REVIEW

File: app/components/cms/community_activity_grid_component/community_activity_grid_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<%= render GovGridRowComponent.new(additional_classes: "community-activity-grid-wrapper") do |row| %>
+  <%= row.with_column(:full) do %>
+    <div class="community-activity-grid">
+      <% if @title %>
```
Comment (A-Wheeto) [inline app/components/cms/community_activity_grid_component/community_activity_grid_component.html.erb:4]: Title is a required parameter in the component, do we need the if statement?

File: app/components/cms/community_activity_grid_component/community_activity_grid_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        <h2 class="govuk-heading-m"><%= @title %></h2>
+      <% end %>
+
+      <%= render @intro.render if @intro %>
```
Comment (A-Wheeto) [inline app/components/cms/community_activity_grid_component/community_activity_grid_component.html.erb:8]: Same question about the intro text, if @intro isn't optional do we need the if?

File: app/components/cms/community_activity_grid_component/community_activity_grid_component_controller.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export default class extends ApplicationController {
+  static values = {
+    createPath: String,
+    slug: String
+  }
+  static targets = ["activityButton", "activityTitle", "activityBox", "activityGrid"];
+
+  selectActivity(event) {
+    console.log(event);
```
Comment (A-Wheeto) [inline app/components/cms/community_activity_grid_component/community_activity_grid_component_controller.js:11]: A console log has snuck through

File: app/webpacker/javascript/controllers/index.js [hunk 4]
```diff
@@ -10,15 +10,22 @@ const contextComponents = require.context("../../../components/", true, /_contro
...
+    // default identifier will be `foo-component--foo-component`. this rewrites those
+    // identifiers to just `foo-component`
+    const isSelfNamed = identifierParts.every(x => x === identifierParts[0])
+    if (!isSelfNamed) return
 
-  definition.identifier = identifierParts[0]
+    definition.identifier = identifierParts[0]
+  }
 })
+console.log(componentDefinitions)
```
Comment (A-Wheeto) [inline app/webpacker/javascript/controllers/index.js:28]: A console log to remove

**Review by A-Wheeto — CHANGES_REQUESTED** [review]
I found a couple of console logs that snuck through and a few of questions on the logic in the community_activity_grid_component, but apart from that looks really good!

File: app/components/cms/community_activity_grid_component/community_activity_grid_component.html.erb (hunk 1 above)
Comment (msquance-stem) [reply app/components/cms/community_activity_grid_component/community_activity_grid_component.html.erb:4]: Both this and the following one are required in the component, but are not required in strapi, so they could come through as nil. I will update the component definition to reflect this better

**Review by A-Wheeto — APPROVED** [review]
Great work! LGTM

**Pull request merged**